### ISSUE TITLE
Feature Request: Network: HTML Labels

### ISSUE THREAD

**Issue by mojoaxel** [almende/vis#1965]
It looks like it is not possible at the moment to create node labels with HTML content. 
In the timeline chart [this is already possible](http://visjs.org/examples/timeline/styling/customCss.html).

see also #973 and #1936

**Comment by mojoaxel** [comment]
Because networks charts are rendered into a canvas this could be done rendering labels into svg elements if they contain html tags.

**Comment by chanb** [comment]
I agree. I've been looking at ways to embed my css into svg elements but have no luck because it's isolated. Would love to have this feature

**Comment by controversial** [comment]
Note that Safari doesn't appear to support HTML in Canvas at all, not even in the latest preview versions. This is a non-standard hack in the first place, iirc.

**Comment by facku8** [comment]
If anyone is facing the issue with Html Labels not working on Safari here is the solution:

Instead: 
`new Blob([url], {type: 'image/svg+xml;charset=utf-8'});`
Replace it by:
`"data:image/svg+xml;utf8,"+ encodeURIComponent(url)`

Hope, it would be helpful and will save you some time...

**Comment by llavayssiere** [comment]
I depend on this feature to improve my visual interface and i hope it will be soon available. It is an essential feature to create node labels with HTML content like "title" node property (docs network : The title can be an HTML element or a string containing plain text or HTML).

**Comment by YakovL** [comment]
May I humbly ask why isn't this implemented if there's [an example](http://visjs.org/examples/network/nodeStyles/HTMLInNodes.html) on the official site? I mean, it would be lovely if this is supported in a manner "[label](http://visjs.org/docs/network/nodes.html): The label is the piece of text **or html** shown in or under the node, depending on the shape." Are there some major problems or just lack of resources to implement?

**Comment by wimrijnders** [comment]
@YakovL I've been considering it as well, actually; the issue I have with this is that it's *not* HTML, just a subset which happens to be supported by `svg`. The next thing that would happen is that everybody starts complaining that their favorite html tag (notably `<a>`!) isn't supported.

In the current way, it's clear that it's not HTML. I kind of like it that way.

**Comment by YakovL** [comment]
@wimrijnders thanks for clarifying, `a` elements are needed indeed, with custom event handlers. Will try ways around.

**Comment by wimrijnders** [comment]
Do you want clickable links specifically?

Hm. That would not be the first request. Perhaps a way to capture clicks on nodes would be in order.
E.g. take a look at #2975. Perhaps there's something useful there.

**Comment by YakovL** [comment]
Well, for now I don't have many use-cases, I'm just adapting vis.js graphs to TiddlyWiki Classic and one obvious desire is to put "wikified" text to nodes, including tiddlyLinks. I'll report back if I find a particular use-case where it's important. Perhaps setting custom click handler to the whole node will be ok in most cases (and I'll see how helpful titles with html may be)

**Comment by cobbs-totem** [comment]
This feature would be extremely useful.  For example, I need to render a class diagram (think UML).  It would be handy to support <hr> for example, to render a horizontal line to separate the class title from the properties.

**Comment by wimrijnders** [comment]
The possibilities for HTML currently are:

- Enable multi-fonts for the nodes. This gives you a grand total of three tags: `<b>`, `<i>` and `<code>`.
- Embed HTML into SVG graphics, see the [example](http://visjs.org/examples/network/nodeStyles/HTMLInNodes.html). The HTML is still limited (e.g. forget about using links and form elements), but it does give more options.

The thing to be aware of, is that `network` draws on a canvas, hence everything you see of it is graphics, not HTML.

I'm not extending HTML any time soon, there are more serious issues to fix - for example, there appears to be one in the example linked above; the text is all scrunchy until you click on a node.

But feel free to contribute! Look into `lib/network/module/components/shared/Label.js` if you're so inclined;  I'm perfectly willing to help you on your way.

**Comment by wimrijnders** [comment]
This issue has been labeled `Inactive`, because there has been no activity here for over two months.

You can:
- Close the issue yourself, so that we know we don't need to take it into account (thanks!).
- Post a new comment to keep it active. The Inactive label and this message will be removed.
- Ignore this comment. If the inactivity continues, the issue will be closed in two weeks.
